yjit/src/codegen.rs

```diff
@@ -2827,6 +2827,7 @@ fn gen_setinstancevariable(
         // generate WB code.
         if !stack_type.is_imm() {
```

Comment:
May not be realistic but I wonder if we could also know at compile time if the "young object" is a frozen string, and skip the wb.

I think regular Ruby strings can have fields eh?

Reply:
I was thinking we might be able to test if it's an `fstring`. We de-dup frozen string literals, and I _think_ they live for the life of the process (but need to double check). In that case we could elide the WB.

Reply:
I am wrong, they don't live forever so we couldn't do this.